### ISSUE TITLE
Does any integrated GPU have enough power for flawless 4K@60 playback?

### ISSUE THREAD

**Issue by 07416** [mpv-player/mpv#9924]
I have [Ryzen 5700G](https://www.amd.com/en/products/apu/amd-ryzen-7-5700g) with 32GB (2 x 16GB) DDR4 3600MHz RAM and I can't find a way to play a 60fps 4K HEVC file smoothly (stray frames are dropped, two at most at once it seems). Enabling hardware acceleration (defaults to `vaapi`) doesn't appear to improve the situation.

I'm only requesting a ballpark estimate and will eventually get a dedicated GPU. Fedora 35 with Gnome on Wayland, mpv from the repo.

[[these 4K 24fps videos](https://github.com/mpv-player/mpv/issues/8981#issuecomment-873605917) dropped no or one frame when playing through each]

[log.txt](https://github.com/mpv-player/mpv/files/8148819/log.txt)

`mediainfo` for the 60fps 4K HEVC file :
```
General
Format                                   : MPEG-4
Format profile                           : Base Media / Version 2
Codec ID                                 : mp42 (isom/iso2/mp41)
File size                                : 10.2 GiB
Duration                                 : 58 min 57 s
Overall bit rate                         : 24.8 Mb/s
Encoded date                             : UTC 2021-12-23 20:09:17
Tagged date                              : UTC 2021-12-23 20:09:17
Writing application                      : HandBrake 20200316165300-6b89f82-master 2020031601

Video
ID                                       : 1
Format                                   : HEVC
Format/Info                              : High Efficiency Video Coding
Format profile                           : Main@L5.1@Main
Codec ID                                 : hvc1
Codec ID/Info                            : High Efficiency Video Coding
Duration                                 : 58 min 57 s
Bit rate                                 : 24.3 Mb/s
Width                                    : 3 840 pixels
Height                                   : 2 160 pixels
Display aspect ratio                     : 16:9
Frame rate mode                          : Variable
Frame rate                               : 59.940 (60000/1001) FPS
Minimum frame rate                       : 59.920 FPS
Maximum frame rate                       : 59.960 FPS
Color space                              : YUV
Chroma subsampling                       : 4:2:0
Bit depth                                : 8 bits
Bits/(Pixel*Frame)                       : 0.049
Stream size                              : 10.0 GiB (98%)
Writing library                          : x265 3.2.1+1-b5c86a64bbbe:[Windows][GCC 9.2.0][64 bit] 8bit+10bit+12bit
Encoding settings                        : cpuid=1111039 / frame-threads=6 / numa-pools=32 / wpp / no-pmode / no-pme / no-psnr / no-ssim / log-level=2 / input-csp=1 / input-res=3840x2160 / interlace=0 / total-frames=0 / level-idc=0 / high-tier=1 / uhd-bd=0 / ref=3 / no-allow-non-conformance / no-repeat-headers / annexb / no-aud / no-hrd / info / hash=0 / no-temporal-layers / open-gop / min-keyint=60 / keyint=600 / gop-lookahead=0 / bframes=4 / b-adapt=0 / b-pyramid / bframe-bias=0 / rc-lookahead=15 / lookahead-slices=8 / scenecut=40 / radl=0 / no-splice / no-intra-refresh / ctu=64 / min-cu-size=8 / no-rect / no-amp / max-tu-size=32 / tu-inter-depth=1 / tu-intra-depth=1 / limit-tu=0 / rdoq-level=0 / dynamic-rd=0.00 / no-ssim-rd / signhide / no-tskip / nr-intra=0 / nr-inter=0 / no-constrained-intra / strong-intra-smoothing / max-merge=2 / limit-refs=3 / no-limit-modes / me=1 / subme=2 / merange=57 / temporal-mvp / no-hme / weightp / no-weightb / no-analyze-src-pics / deblock=0:0 / sao / no-sao-non-deblock / rd=2 / selective-sao=4 / no-early-skip / rskip / fast-intra / no-tskip-fast / no-cu-lossless / no-b-intra / no-splitrd-skip / rdpenalty=0 / psy-rd=2.00 / psy-rdoq=0.00 / no-rd-refine / no-lossless / cbqpoffs=0 / crqpoffs=0 / rc=crf / crf=20.0 / qcomp=0.60 / qpstep=4 / stats-write=0 / stats-read=0 / ipratio=1.40 / pbratio=1.30 / aq-mode=2 / aq-strength=1.00 / cutree / zone-count=0 / no-strict-cbr / qg-size=32 / no-rc-grain / qpmax=69 / qpmin=0 / no-const-vbv / sar=1 / overscan=0 / videoformat=5 / range=0 / colorprim=1 / transfer=1 / colormatrix=1 / chromaloc=0 / display-window=0 / cll=0,0 / min-luma=0 / max-luma=255 / log2-max-poc-lsb=8 / vui-timing-info / vui-hrd-info / slices=1 / no-opt-qp-pps / no-opt-ref-list-length-pps / no-multi-pass-opt-rps / scenecut-bias=0.05 / no-opt-cu-delta-qp / no-aq-motion / no-hdr / no-hdr-opt / no-dhdr10-opt / no-idr-recovery-sei / analysis-reuse-level=5 / scale-factor=0 / refine-intra=0 / refine-inter=0 / refine-mv=1 / refine-ctu-distortion=0 / no-limit-sao / ctu-info=0 / no-lowpass-dct / refine-analysis-type=0 / copy-pic=1 / max-ausize-factor=1.0 / no-dynamic-refine / no-single-sei / no-hevc-aq / no-svt / no-field / qp-adaptation-range=1.00
Encoded date                             : UTC 2021-12-23 20:09:17
Tagged date                              : UTC 2021-12-23 20:09:17
Color range                              : Limited
Color primaries                          : BT.709
Transfer characteristics                 : BT.709
Matrix coefficients                      : BT.709
Codec configuration box                  : hvcC
```

**UPDATE**: I didn't take HDR into consideration since there's no support on the Linux desktop yet — I assumed that any SDR tone-mapping isn't significantly more resource-intensive. The 4K@60 video isn't in HDR.

**Comment by wsldankers** [comment]
I'm able to play Gemini Man (4K@59.94) using an AMD Ryzen 2400G, using vaapi and with debanding disabled. No frame drops. This is using `--gpu-context=drm` though.

**Comment by philipl** [comment]
I can't speak for Ryzen, but no Intel iGPU can do 4k60 HDR (or really any kind of advanced shader processing). There is _just_ enough power to do 4k24 with HDR. This isn't about decode capability (why vaapi didn't help) but the GPU itself.

**Comment by 07416** [comment]
> I'm able to play Gemini Man (4K@59.94) using an AMD Ryzen 2400G, using vaapi and with debanding disabled. No frame drops. This is using `--gpu-context=drm` though.

`--gpu-context=drm` appears to run dreadfully on my system, but I have a graphical environment (Gnome) installed.

I had the film at hand, 20 frames dropped at the two minute mark. `--no-config  --deband=no` doesn't seem to have an impact, `--hwdec=vaapi` appears to run clearly worse. I don't have a HDR screen and allow mpv to perform SDR tone-mapping.

```
Codec: | h265 (HDR 10-bit)
Resolution: | 3840x2160
Aspect ratio: | 16:9
Frame rate: | 59.940 fps
Bit rate: | 81.0 Mb/s
```

**Comment by wsldankers** [comment]
My mpv.conf is geared towards “high quality” (`fbo-format=rgba16f`, `drm-format=xrgb2101010`, ICC profile, etcetera but the display is 4K as well, so no scaling). Like your case it also has to do tone-mapping (but I disabled peak detection: `--hdr-compute-peak=no`). Your processor is several generations newer, so I'd expect it to keep up easily.

The biggest difference is probably that my system is completely pared down for movie watching, there's no graphical desktop environment whatsoever. It might be that the compute budget for each frame is a bit tight and the desktop environment interfering sometimes causes it to overrun.

**Comment by 07416** [comment]
> Display resample causes framerates to drop by ~50% for me and I don't think that insane cache amount you set is a good idea. Maybe try without them or better yet, see what playback is like without any extra config or scripts running.

I only recently added `video-sync=display-resample` and chose to retain after seeing no immediate drawbacks. The cache setting was a leftover from testing and I've mostly used the defaults in the past (incl. any other more complex options). 

I'll resubmit the logs with stock options and other videos. I also plan to try without a Linux DE.

**Comment by 07416** [comment]
I reseted the MSI B550 Tomahawk iGPU settings in BIOS, I had modified the 'UMA buffer size' (maxed out at 16GB) since there appeared to be a boost in gaming performance compared to automatic. I also disabled some to me unnecessary features such as on-board audio and RGB lighting, but assume that the former are unlikely to have any clear effect.

`Gemini.Man.2019.UHD.BluRay.2160p.TrueHD.Atmos.7.1.HEVC.remux.mkv` now dropped no frames after playing for around five minutes, with `display-resample`, only about 200 delayed (frames). I'll have to see if 24P playback is also problem-free now (another topic).
